src/vm/functions/conversions.rs

```diff
@@ -113,3 +115,107 @@ pub fn native_buff_to_uint_be(value: Value) -> Result<Value> {
     }
     return buff_to_int_generic(value, EndianDirection::BigEndian, convert_to_uint_be);
 }
+
+pub fn native_string_to_int_generic(
+    value: Value,
+    conversion_fn: fn(String) -> Result<Value>,
+) -> Result<Value> {
+    match value {
+        Value::Sequence(SequenceData::String(CharType::ASCII(ASCIIData { data }))) => {
+            let as_string = String::from_utf8(data).unwrap();
+            return conversion_fn(as_string);
+        }
+        Value::Sequence(SequenceData::String(CharType::UTF8(UTF8Data { data }))) => {
+            let flat = data.into_iter().flatten().collect();
+            let as_string = String::from_utf8(flat).unwrap();
+            return conversion_fn(as_string);
+        }
+        _ => {
+            return Err(CheckErrors::UnionTypeValueError(
+                vec![
+                    TypeSignature::max_string_ascii(),
+                    TypeSignature::max_string_utf8(),
+                ],
+                value,
+            )
+            .into())
+        }
+    };
+}
+
+pub fn native_string_to_int(value: Value) -> Result<Value> {
+    fn convert_to_int(raw_string: String) -> Result<Value> {
+        let possible_int = raw_string.parse::<i128>();
+        match possible_int {
```

Comment:
This `match` statement really should not return an `Err(..)`, because the user may submit a code snippet like `(string-to-uint "this is definitely not an integer")`.  An `InterpreterError` will cause the transaction to be invalid, which is not what we want.

What we really want is for `(string-to-uint)` and `(string-to-int)` to return either `(optional int|uint)` or `(response int|uint int)` -- i.e. a type that represents a possibly-successful but not guaranteed-successful conversion of the string type to an integer type.  I'm not too picky on `optional` vs `response`, but the function must evaluate to either a "success" or "failed" state that the caller can handle.

As to what this means for this `match` statement, you should be returning `Value::some(Value::Int(val))` or `Value::none()`, or `Value::okay(Value::Int(val))` or `Value::error(Value::Int(error_code))`.  Same goes for `native_string_to_uint` below.

Reply:
i've gone with the "optional approach", and changed the documentation and tests accordingly.. what do you think?

Reply:
Works for me!